soe/dynamic_test.go

```diff
@@ -0,0 +1,206 @@
+package soe_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hamba/avro/v2"
+	"github.com/hamba/avro/v2/soe"
+	"github.com/hamba/avro/v2/soe/testdata"
+	"github.com/stretchr/testify/require"
+)
+
+// Helper function to marshal v into an SOE-framed binary Avro encoding.
+func encode(t *testing.T, schema avro.Schema, v any) []byte {
+	t.Helper()
+
+	// Use a basic codec to marshal value.
+	codec, err := soe.NewCodec(schema)
+	require.NoError(t, err)
+
+	data, err := codec.Encode(&v)
+	require.NoError(t, err)
+	return data
+}
+
+// Helper function to create a new DynamicDecoder with a registry over zero or
+// more schemas.
+func newDynamicDecoder(t *testing.T, schemas ...avro.Schema) *soe.DynamicDecoder {
+	t.Helper()
+
+	// Set up a registry with provided schemas.
+	registry := soe.NewMemorySchemaRegistry()
+	for _, schema := range schemas {
+		err := registry.AddSchema(schema)
+		require.NoError(t, err)
+	}
+	return soe.NewDynamicDecoder(registry)
+}
+
+func TestDynamicDecodeUnknownSchema(t *testing.T) {
+	// Marshal a value
+	data := encode(t, testdata.Dynamic1Schema, testdata.Dynamic1{
+		Name: "Bob",
+		Age:  16,
+	})
+
+	// Set up a decoder with an empty registry.
+	decoder := newDynamicDecoder(t)
+
+	// Decode should fail with unknown schema error.
+	var v1 testdata.Dynamic1
+	err := decoder.Decode(context.Background(), data, &v1)
+	require.ErrorIs(t, err, soe.ErrUnknownSchema)
+}
+
+func TestDynamicDecodeOneSchema(t *testing.T) {
```

Comment:
Conventionally the test naming would be `TestDynamicDecoder_Decode<test scenario>`